nntrainer/utils/node_exporter.h

```diff
@@ -106,24 +145,83 @@ class Exporter {
 
   /**
    * @brief Get the result object
+   * @note @a unique_ptr here will be a member of @a this. The ownership is
+   * passed to the caller, thus after getResult, the target member is cleared.
+   * This is intended design choice to mimic single function call, between @a
+   * saveResult and @a getResult
    *
    * @tparam methods method to get
    * @tparam T appropriate return type regarding the export method
-   * @return T T
+   * @return std::unique_ptr<T> predefined return type according to the method.
+   * @retval nullptr not exported
    */
   template <ExportMethods methods,
             typename T = typename return_type<methods>::type>
-  const T &get_result();
+  std::unique_ptr<T> getResult() noexcept;
 
 private:
-  std::vector<std::pair<std::string, std::string>>
+  /**
+   * @brief ProtoType to enable saving result to tfnode
+   *
+   * @tparam PropsType propsType PropsType to receive with self
+   * @tparam NodeType NodeType of current layer.
+   * @param props properties to get with @a self
+   * @param self @a this of the current layer
+   */
+  template <typename PropsType, typename NodeType>
+  void saveTflResult(const PropsType &props, const NodeType *self);
+
+#ifdef ENABLE_TFLITE_INTERPRETER
+  std::unique_ptr<TfOpNode> tf_node; /**< created node from the export */
+#endif
+
+  std::unique_ptr<std::vector<std::pair<std::string, std::string>>>
     stored_result; /**< stored result */
 
   /// consider changing this to a promise / future if there is a async function
-  /// involved to `save_result`
+  /// involved to `saveResult`
   bool is_exported; /**< boolean to check if exported */
 };
 
+/**
+ * @brief ProtoType to enable saving result to tfnode
+ *
+ * @tparam PropsType propsType PropsType to receive with self
+ * @tparam NodeType NodeType of current layer.
+ * @param props properties to get with @a self
+ * @param self @a this of the current layer
+ */
+template <typename PropsType, typename NodeType>
+void Exporter::saveTflResult(const PropsType &props, const NodeType *self) {
+  NNTR_THROW_IF(true, nntrainer::exception::not_supported)
+    << "given node cannot be converted to tfnode";
+}
+
+#ifdef ENABLE_TFLITE_INTERPRETER
+namespace props {
+class Name;
+class Unit;
+} // namespace props
+
+class Layer;
+/**
+ * @copydoc template <typename PropsType, typename NodeType> void
+ * Exporter::saveTflResult(const PropsType &props, const NodeType *self);
+ */
+template <>
+void Exporter::saveTflResult(const std::tuple<props::Name> &props,
+                             const Layer *self);
+
+class FullyConnectedLayer;
+/**
+ * @copydoc template <typename PropsType, typename NodeType> void
+ * Exporter::saveTflResult(const PropsType &props, const NodeType *self);
+ */
+template <>
+void Exporter::saveTflResult(const std::tuple<props::Unit> &props,
+                             const FullyConnectedLayer *self);
+#endif
```

Comment:
It will be bigger than this. I think it would be better to manage these functions in a separate file. then we could remove ifdef in the code as well.

Reply:
Yes, I'll make sure this be a seperate file :)